### PR TITLE
Init conda after installing requirements in tests

### PR DESCRIPTION
**Pull request by jjhelmus**
### Description

When setting up to run the unit tests initialize conda only after the requirements are installed since installing requirements may itself install conda undoing any initialization.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by jjhelmus** [comment]
The ordering of these operations is the cause of the failure in #12438.

**Review by jezdez — APPROVED** [review]

**Comment by jezdez** [comment]
Nice find!

**Comment by dholth** [comment]
Will this make Windows use the github actions cache of packages more effectively?

File: dev/linux/unit.sh [hunk 1]
```diff
@@ -8,9 +8,9 @@ git config --global --add safe.directory /opt/conda-src
 TEST_SPLITS="${TEST_SPLITS:-1}"
 TEST_GROUP="${TEST_GROUP:-1}"
 
-eval "$(sudo /opt/conda/bin/python -m conda init --dev bash)"
 # make sure all test requirements are installed
 sudo /opt/conda/bin/conda install --quiet -y -c defaults --file tests/requirements.txt
```
Comment (jaimergp) [inline dev/linux/unit.sh:12]: This line shouldn't be adding `defaults`. I'll open an issue and a PR.

**Pull request merged**